### PR TITLE
feat(deliveryFunctions): cca build and upload [MONET-1336]

### PR DESCRIPTION
**Pull request by kdamball**

### CODE REVIEW

File: packages/contentful--app-scripts/lib/clean-up/clean-up-bundles.test.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const bundlesFixture = [
   { sys: { id: 'test-8' } },
 ];
 
-describe.only('cleanUpBundles', () => {
```
Comment (ghepting) [inline packages/contentful--app-scripts/lib/clean-up/clean-up-bundles.test.js:17]: oof lol

File: packages/contentful--app-scripts/lib/utils.js [hunk 2]
```diff
@@ -125,8 +125,68 @@ function getActionsManifest() {
...
+
+    if (!Array.isArray(manifest.deliveryFunctions) || manifest.deliveryFunctions.length === 0) {
+      return;
+    }
+
+    console.log('');
+    console.log(`  ----------------------------
+  Delivery functions found in ${chalk.bold(DEFAULT_MANIFEST_PATH)}.
+  ----------------------------`);
+    console.log('');
```
Comment (ghepting) [inline packages/contentful--app-scripts/lib/utils.js:154]: we should really put this formatting in a logging function, huh?

Comment (kdamball) [reply packages/contentful--app-scripts/lib/utils.js:154]: Yep - I remember we have something like this already somewhere

**Comment by kdamball** [comment]
@BasKiers now I have all your commits 🙈

File: packages/contentful--app-scripts/package.json [hunk 3]
```diff
@@ -27,7 +27,7 @@
     "prettier": "prettier **/*.js --write --ignore-path .gitignore",
     "lint": "eslint ./lib",
     "lint:fix": "npm run lint -- --fix",
-    "test": "mocha ./lib/**/*.test.js ./lib/*.test.js ./utils/**/*.test.js",
+    "test": "mocha \"./{,!(node_modules)/**/}*.test.js\" --exit",
```
Comment (andipaetzold) [inline packages/contentful--app-scripts/package.json:30]: Why this change?

File: packages/contentful--app-scripts/lib/utils.js [hunk 4]
```diff
@@ -125,8 +127,64 @@ function getActionsManifest() {
...
-      )}.`
+        DEFAULT_MANIFEST_PATH,
+      )}.`,
+    );
+    // eslint-disable-next-line no-process-exit
+    process.exit(1);
+  }
+}
+
+function getDeliveryFunctionsManifest() {
```
Comment (andipaetzold) [inline packages/contentful--app-scripts/lib/utils.js:138]: Could we somehow merge `getActionsManifest` and `getDeliveryFunctionsManifest` methods? They are both parsing the same file. When reading this code first, I was confused why we now have two manifests.

File: packages/contentful--app-scripts/lib/utils.js [hunk 5]
```diff
@@ -125,8 +127,64 @@ function getActionsManifest() {
...
+            'Error:',
+          )} Invalid IP address ${hasInvalidNetwork} found in the allowNetworks array for delivery function "${
+            deliveryFn.name
+          }".`,
+        );
+        // eslint-disable-next-line no-process-exit
+        process.exit(1);
+      }
+
+      // EntryFile is not used but we want to strip it from delivery functions
```
Comment (andipaetzold) [inline packages/contentful--app-scripts/lib/utils.js:172]: grammar
```suggestion
      // EntryFile is not use, so we want to strip it from delivery functions
```

File: packages/contentful--app-scripts/package.json (hunk 3 above)
Comment (kdamball) [reply packages/contentful--app-scripts/package.json:30]: previous one was weird because every time you add a new folder you'd need to remember to pass the folder in here. I opted to just run everything with the `.test.js` suffix and also exit the test runner as previously we weren't even running them in the first place 🤦🏾

Comment (andipaetzold) [reply packages/contentful--app-scripts/package.json:30]: The trick (I think) for mocha is to use quotes. Running `mocha "path/**/file.js"` is different to `mocha path/**/file.js`.

This also does the job and is maybe easier to understand as it doesn't have negation: 
```
mocha \"./{lib,utils}/**/*.test.js\" --exit
```

Comment (kdamball) [reply packages/contentful--app-scripts/package.json:30]: That's exactly the problem I was trying to avoid - forgetting add new folders here because I'm pretty sure it will happen again 😅 . I'd be in favour of moving `utils` folder into the `lib` but that's probably a discussion for another PR

Comment (andipaetzold) [reply packages/contentful--app-scripts/package.json:30]: +1 for only having one source folder.
I guess I have the same discussion in another PR atm: https://github.com/contentful/create-contentful-app/pull/1634#discussion_r1302874023

File: packages/contentful--app-scripts/lib/utils.js [hunk 6]
```diff
@@ -73,7 +79,7 @@ const selectFromList = async (list, message, cachedOptionEnvVar) => {
   }
 };
 
-function getActionsManifest() {
+function getEntityFromManifest(type) {
```
Comment (andipaetzold) [inline packages/contentful--app-scripts/lib/utils.js:82]: Alternatively, we could always parse both and return an object with two props 🤷

**Review by andipaetzold — APPROVED** [review]

File: packages/contentful--app-scripts/lib/utils.js (hunk 6 above)
Comment (kdamball) [reply packages/contentful--app-scripts/lib/utils.js:82]: But this way the function is smaller 😛
